backend/Dockerfile

```diff
@@ -5,13 +5,16 @@ FROM python:3.12
 WORKDIR /backend
 
 # Copy just the requirements into the working directory so it gets cached by itself
-COPY ./requirements.txt ./requirements.txt
+COPY backend/requirements.txt ./requirements.txt
 
 # Install the dependencies from the requirements file
 RUN pip install --no-cache-dir --upgrade -r /backend/requirements.txt
 
 # Copy the source code into the working directory
-COPY ./src/. ./src
+COPY backend/src/. ./src
+
+# Copy the env.yaml file from the root directory to the backend directory
+COPY env.yml ./env.yml
```

Comment:
why does the env file have to move to the backend directory?

Reply:
It's because the docker is set up for backend to copy all the files before initialising the project, earlier as we were using .env which gets auto loaded as an environment file and made available using dotenv library. There is no similar library to load yml files as such, so I have to manually move it to the backend folder otherwise the config file is unable to find it.